maven-surefire-plugin/src/site/apt/examples/inclusion-exclusion.apt.vm

```diff
@@ -164,7 +164,16 @@ Inclusions and Exclusions of Tests
 +---+
 
   Note the syntax <<<%regex[expr]>>>, where <<<expr>>> is the actual expression and the rest is just wrapping. Also
-  note that regex matches are done over <<<*.class>>> files and not <<<*.java>>> files.
+  note the following about the use of regular expressions:
+
+   * Regex matches are done over <<<*.class>>> files and not <<<*.java>>> files
+
+   * Regex matches are done over paths using slashes ("<<</>>>") and not package names using dots ("<<<.>>>"), so the
+      "<<<.>>>" in <<<pkg.*Slow.*.class>>> is a regex metacharacter, which happens to match any character, notably
+      the (forward) slashes ("<<</>>>") that make up the path. Slashes here are <forward>, even on Windows
```

Comment:
We have a plan to discontinue this old format in Version `3.0.0` and use only `fully qualified class name` w/o regex. It does not make sense to maintain current feature. It is complicated since we have to use the filter twice, on file system and then in JUnit/TestNG f/w. In the future all paths will be just classes (with dot `.`) and then the filter would work on file systems and test frameworks.

Reply:
How will that approach work for being able to specify packages? When unit tests run, we don't want to run anything with an `it` package at its root. Having to specify fully qualified classnames for every single integration test to not run in the Surefire configuration and then either specify every integration test to run in Failsafe or every unit test not to run would be a configuration nightmare.

We had been using `it/**` until we upgraded Surefire/Failsafe, but now that has a behavior of automatically prepending `**/` to any pattern that doesn't start with it, which ignores any class with an `it` package anywhere, rather than just at the root. We switched to `%regex` to work around that change.

Does the new solution had some sort of wildcard syntax planned? And, if so, will it have a way to say we don't want any "helpful" modifications to what we configure?

Reply:
@bturner 
It will run the same as before except that there would not be slashes and `.class` nor `.java`.
We have a huge number of unit and integration tests. So. We try to pass them successfully.

Reply:
@bturner 
You can track the progress in [SUREFIRE-1292](https://issues.apache.org/jira/browse/SUREFIRE-1292) and [SUREFIRE-526](https://issues.apache.org/jira/browse/SUREFIRE-526).